### DISCUSSION TITLE
'select' not available for AsyncNotifierProvider?!

### DISCUSSION THREAD

**Discussion by davoutuk** [rrousselGit/riverpod#2618] [General]
I've started to review RiverPod 2.0 and the use of AsyncNotifierProvider...,

It seems that the 'select' method of retrieving specific elements of the notifier model is not available for this type of provider, why is that?

**Comment by AhmedLSayed9** [comment]
For async providers there's a [selectAsync](https://pub.dev/documentation/riverpod/latest/riverpod/AutoDisposeFutureProvider/selectAsync.html) which is a variant of select for asynchronous values.

**Reply by davoutuk** [reply]
Looking at the API docs, there is no 'select' or 'selectAsync', see: https://pub.dev/documentation/riverpod/latest/riverpod/AsyncNotifierProviderFamily-class.html

**Reply by rrousselGit** [reply]
You're looking at the "family" class. Using it, you're meant to pass parameters: `ref.watch(family(param).select(...))`

**Reply by davoutuk** [reply]
OK...  would it be possible to upgrade the API docs/tutorial with an example?
